### PR TITLE
refactor(icons): revamp icons for ECL 2.x

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Improve the way icons are managed in ECL.

TL;DR: Icons now uses svg files instead of icons font. All svgs are merged in an "svg sprite" file.

**File structure**
- Social media icons have been moved out of component, and put in a global `resources` folder (one per system)
- Previous icons (in a font) have been put in resources folder too (as svg)

**Components**
- all components using icons now use svg images (no more background images or font icons)

**Internet Explorer support**
- Third part library is needed to display icons correctly on Internet Explorer 11.
See icons readme file for more information.

**Additional information**

- Legacy font icons has been kept for now, for backward compatibility, but should be considered as deprecated.
- Screenshots have not been updated as they will change a lot in upcoming days; we will update them once branch next-v2 is more stable.

Further improvements
- Harmonize variable name for icons, and put them in an array for each component
- Update templates

Follow-up ticket: https://webgate.ec.europa.eu/CITnet/jira/browse/INNO-1135

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit a0bfefaaeddf0eaa8257da2711293e8d556ec706

https://deploy-preview-726--europa-component-library.netlify.com

File: src/systems/ec/ec-resources/icons/ui/if_gear_298781.svg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<!-- Generated by IcoMoon.io -->
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="28" height="32" viewBox="0 0 28 32">
+<title>if_gear_298781</title>
```
Comment (raduchiriac) [inline src/systems/ec/ec-resources/icons/ui/if_gear_298781.svg]: Can we rename this `cog` as per https://webgate.ec.europa.eu/CITnet/jira/browse/INNO-953 or more preferably `gear`?

File: src/systems/eu/eu-style/eu-style-icon/eu-style-icon.config.js [hunk 2]
```diff
@@ -2,22 +2,17 @@ module.exports = {
...
-          item.context.modifier
-        }"></span>\n
-        <span class="ecl-icon ecl-u-color-primary ${
-          item.context.modifier
-        }"></span>\n
-        <code>${item.context.modifier}</code>\n
-      </p>\n
-      <!-- End: @${item.handle} -->\n
+    <!-- Start demo of: @${item.handle} -->\n
+    <div class="ecl-u-d-inline-flex">
```
Comment (raduchiriac) [inline src/systems/eu/eu-style/eu-style-icon/eu-style-icon.config.js]: What about a `.ecl-u-align-items-center` here?

**Comment by raduchiriac** [comment]
Everything looks fine from what I could test. Good work :+1: 

Things we don't have to forget about:
- future of `generic-style-icon/fonts`
- future of `ec-resources/icons/social/` and `/ui`

**Review by raduchiriac — CHANGES_REQUESTED** [review]
Can you please fix this last thing? `error  Unable to resolve path to module '@ecl/generic-component-button/data/demo'  import/no-unresolved`
